stablehlo/transforms/ChloDecompositionUtils.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2024 The StableHLO Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_TRANSFORMS_CHLO_DECOMP_UTILS_H_
+#define STABLEHLO_TRANSFORMS_CHLO_DECOMP_UTILS_H_
+
+#include "mlir/IR/Value.h"
+#include "mlir/IR/ValueRange.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace stablehlo {
+
+// Utility functions used in the Chlo to stablehlo legalization.
+
+Value materializeLgamma(ConversionPatternRewriter &rewriter, Location loc,
```

Comment:
Could these be generalized to take a patternrewriter or even opbuilder?

This will be necessary to use the functions outside a conversion

Reply:
+1 I was wondering what ideal way of exposing things like this would be. Looking at the impl, there's no need for this to be a pattern rewriter (I don't think), seems to only be using the `create` method from the `OpBuilder` base class. Could flip to use that.

Reply:
I just did a `static_cast` to convert it to `ConversionPatternRewriter` atm, but I also don't think anything specific to dialect conversion is being called.

Reply:
Could we instead change all interfaces `ConversionPatternRewriter -> OpBuilder`? _I think_ it should just work with a find/replace.

That's be the most generic entrypoint, i.e. "This will be necessary to use the functions outside a conversion"

Reply:
in general the static up cast is invalid because we may call it with something which isn't a `ConversionPatternRewriter` and static casting and pretending may result in issues. Just making it take an `OpBuilder` fixes this

Reply:
I think we are good to merge now - I converted almost everything to OpBuilder, and all the tests pass